src/main/java/pl/project13/core/NativeGitProvider.java

```diff
@@ -489,12 +486,9 @@ public boolean runEmpty(File directory, long nativeGitTimeoutInMs, String comman
       final AtomicBoolean empty = new AtomicBoolean(true);
 
       try {
-        final Function<String, Boolean> stdoutConsumer = line -> {
-          if (line != null) {
-            empty.set(false);
-          }
-          // return false to indicate we don't need to read more content
-          return false;
```

Comment:
Removed this logic to stop reading from the output. It is always necessary to read from the output, even if the lines are then discarded, otherwise the output buffer can get full and the process gets stuck writing to stdout.